AGENTS.md

```diff
@@ -0,0 +1,443 @@
+cp .env.sample.openai .env
+# Edit .env and set:
+# API_HOST=openai
+# OPENAI_API_KEY=your_openai_api_key
+# OPENAI_MODEL=gpt-4o-mini
+```
+
+#### Option 4: Ollama (requires local Ollama installation)
+
+1. Install Ollama: https://ollama.com/
+2. Pull a model:
+   ```bash
+   ollama pull llama3.1
+   ```
+3. Create .env:
+   ```bash
+   cp .env.sample.ollama .env
+   # Edit .env and set:
+   # API_HOST=ollama
+   # OLLAMA_ENDPOINT=http://localhost:11434/v1
+   # OLLAMA_MODEL=llama3.1
+   ```
+   **Important:** If running in a Dev Container, use `http://host.docker.internal:11434/v1` instead of `localhost`.
+
+### Running Scripts
+
+After environment setup and .env configuration, run any script:
+
+```bash
+python chat.py
+python function_calling_basic.py
+python rag_csv.py  # Requires requirements-rag.txt to be installed
+```
+
+For Spanish versions:
+```bash
+python spanish/chat.py
+```
+
+## Running Tests
+
+### Linting with Ruff
+
+```bash
+# After installing requirements-dev.txt
+ruff check .
+```
+
+To auto-fix issues:
+```bash
+ruff check . --fix
+```
+
+### Formatting with Black
+
+Check formatting:
+```bash
+black . --check --verbose
+```
+
+Auto-format:
+```bash
+black .
+```
+
+### Pre-commit Hooks
+
+Install pre-commit hooks to automatically run checks before commits:
+```bash
+pre-commit install
+```
+
+Run manually:
+```bash
+pre-commit run --all-files
+```
+
+### Integration Tests
+
+The repository has limited automated testing via GitHub Actions. The primary test runs basic scripts with GitHub Models:
+
+```bash
+# This is what the CI does (requires GitHub token):
+export API_HOST=github
+export GITHUB_TOKEN=$YOUR_TOKEN
+export GITHUB_MODEL=openai/gpt-4o-mini
+python chat.py
+python spanish/chat.py
+```
+
+**Note:** Most scripts are demonstration scripts, not unit-tested. Changes to scripts should be manually verified by running them.
+
+## Upgrading Python Dependencies
+
+This repository uses **standard pip** for dependency management (not poetry, pipenv, or uv for installation).
+
+### To upgrade a single dependency:
+
+```bash
+# Activate your virtual environment first
+source .venv/bin/activate
+
+# Upgrade a specific package
+python -m pip install --upgrade package-name
+
+# Update requirements file manually to reflect the new version
+```
+
+### To upgrade all dependencies:
+
+```bash
+# Activate your virtual environment
+source .venv/bin/activate
+
+# Upgrade all packages
+python -m pip install --upgrade -r requirements.txt
+python -m pip install --upgrade -r requirements-rag.txt
+
+# Freeze to see new versions (optional)
+python -m pip freeze
+```
+
+### Important Notes:
+
+1. **The CI uses `uv` for speed**, but you should use standard `pip` for local development unless you have `uv` installed.
+2. After upgrading dependencies, always test by running the linters and at least a few example scripts.
+3. The `openai` package has a minimum version constraint: `openai>=1.108.1` in requirements.txt. Don't downgrade below this.
```

Comment:
Make this comment more general. If a version is pinned, the versio should never be lowered.